sdk/resource/config.go

```diff
@@ -67,7 +55,24 @@ func (d detectAttributes) Detect(context.Context) (*Resource, error) {
 }
 
 // WithDetectors adds detectors to be evaluated for the configured resource.
+// Any use of WithDetectors disabled the default behavior, to reenable this
+// inlcude a WithDetectors(BuiltinDetectors...),
+// Examples:
+// `New(ctx)`: Use builtin `Detector`s.
+// `New(ctx, WithDetectors())`: Use no `Detector`s
+// `New(ctx, WithDetectors(NoOp{}))`: Use no `Detector`s
+// `New(ctx, WithDetectors(d1, d2))`: Use Detector `d1`, then overlay Detector `d2`
```

Comment:
```suggestion
// `New(ctx, WithDetectors())`: Use no `Detector`s.
// `New(ctx, WithDetectors(NoOp{}))`: Use no `Detector`s.
// `New(ctx, WithDetectors(d1, d2))`: Use Detector `d1`, then overlay Detector `d2`.
```
If `WithDetectors` with no arguments does the same thing as passing `NoOp`, do we still need to export `NoOp`?